### ISSUE TITLE
An unhandled error has occurred. Reload

### ISSUE THREAD

**Issue by maviscass** [Xeio/WanderLost#77]
The site have been asking me to reload for the past 6 hours. I cleared cookies, cache and history, it is still the same. 
The timer will continue to count down but I won't be able to see any new submission nor able to submit anything.

**Comment by Xeio** [comment]
Hmmm, interesting, if you see timers count down that means at least all the assemblies should have loaded. Likely means there was some issue connecting to the server for the actual merchants (though I'm guessing it was working before?).

Assuming this is a PC, can you hit F12 in your browser and open the dev tools panel, it should look something like this if you select the "Console" tab circled at the top:

![errors console](https://user-images.githubusercontent.com/25418/166239175-a3d33ad4-ff17-4743-b27c-4b5a3ef2db67.png)

Basically looking for the big blob of red text that should be toward the bottom.

**Comment by maviscass** [comment]
It somehow got fixed after restarting the computer, I should have done that
first. Yes it is on PC. If this happens again then I will do the F12 thing.

Thank you so much for prompt reply and very good work on the website!

**Comment by maviscass** [comment]
Hello, so it happened again after I clicked "Log in on Discord"
[image: image.png][image: image.png][image: image.png][image: image.png]

**Comment by Xeio** [comment]
None of those images seem to have come through

**Comment by maviscass** [comment]
Hi, so after logging out and refreshed, it works fine.

[image: image.png]

**Comment by Xeio** [comment]
If you're replying via-email I think images don't work like that, just so you know.

**Comment by maviscass** [comment]
![image (3)](https://user-images.githubusercontent.com/104755792/166289129-f6ebd60e-df81-49b7-8518-9dea99879900.png)
![image (2)](https://user-images.githubusercontent.com/104755792/166289136-bd17888d-9812-460e-b5cf-8d97c1c5a432.png)
![image (1)](https://user-images.githubusercontent.com/104755792/166289141-156bca7e-a869-478f-b1a5-871d0d9ad70e.png)
![image](https://user-images.githubusercontent.com/104755792/166289146-545ecd8d-eaaf-4d86-a273-7c9bdee3d2c9.png)

This is after I login with Discord. 

After logging out then it works fine.

**Comment by Xeio** [comment]
Hmm, so if you see it again you may want to click this arrow (circled in orange) to expand it:

![Error](https://user-images.githubusercontent.com/25418/166289881-e3de0907-5c50-4711-844d-9a285a3e4ef6.png)

It says the connection failed, though if there's any more detail it's hidden

**Comment by Xeio** [comment]
Huh... that's... not helpful at all. No failure message which is strange.

Shouldn't be any notable changes in the last day for this either, and it's odd that it seems intermittent.

One last thing you could try is in that same dev tools page, click the "Network" tab at the top and look for the "MerchantHub" request (you may have to reload the page while the dev tools are open for it to show up). Also there's a lot of junk there so I recommend typing into the search box up top so it filters (though you can just find it in the list maybe).

![image](https://user-images.githubusercontent.com/25418/166296196-bc38921c-0bf5-4537-9856-a315e989a0ba.png)

Though unless that has some useful error info I'm not sure what could be happening.

**Comment by Xeio** [comment]
One minor thing, I did tweak the "Reload" button on the error panel to purge any cached data on the site after an exception. Probably won't resolve your issue but may help some others.

**Comment by maviscass** [comment]
Two friends of mine having the same issue as me, I told them to log out of discord and it auto refresh for them now.

**Comment by bdurrer** [comment]
Just a FYI, because I didn't read anything about what clients were being used

- I had that error message on the _mobile_ Chrome yesterday. Tested again today - works just fine.
- Mobile Firefox worked. 
- Desktop FF and Chrome always worked for me and have no errors in console, on multiple machines, both logged in and anonymous

**Comment by jeremy-hill** [comment]
Also getting this in Chrome Version 101.0.4951.54 (Official Build) (64-bit). Website was working in the past, no longer however.

![image](https://user-images.githubusercontent.com/66854273/167467825-f579469d-4601-4870-9919-77a4aefb27fd.png)

**Comment by Xeio** [comment]
Yeah, seeing some significant pickup in error logs today, not just your end.

I may try restarting everything when I have a moment, but I'm currently working so can't do much digging as to the cause.

**Comment by Xeio** [comment]
It _seems_ like it's holding with all the clients re-connected after restarting all the containers, and I can refresh/login/logout without issues which was failing a lot earlier.

Going to have to see if any of the logs captured help diagnose what happened after work.

**Comment by Rhythmslc** [comment]
I have this issue also but I was also having issues before usually once or twice a day I have to reload to see merchants.  not there is a error that's more helpful but if it would have a auto reload on this failed state would be preferred.

**Comment by Xeio** [comment]
> I have this issue also but I was also having issues before usually once or twice a day I have to reload to see merchants

Specifically for that there was a bug I fixed on Sunday around that so it hopefully shouldn't happen anymore, had to do with the client reconnection process. You wouldn't receive any server push updates without reloading the page.

Generally the client already tries to reconnect for connectivity issues, but reloading would probably just end up with clients going into infinite loops if there's some other underlying issue.

**Comment by Rhythmslc** [comment]
I haven't had to force reload since the change very nice and fast fix.

**Comment by wolfgangjt** [comment]
still broken, has been constantly loosing connection, reload, buttons not working, etc

**Comment by Xeio** [comment]
It's not still broken, I broke it in a new and different way this time after messing with the database because I can't leave well enough alone. 💀 

Should be back up right about now though.

**Comment by ytzyt** [comment]
Seems like a server problem? I've seen this reload thing many times too.

Just now I had a different warning after I took the F12 red text screen shots

![P1](https://user-images.githubusercontent.com/9277616/168450961-d76dcc00-e2c3-448c-971c-dd45e501b34c.png)
![P2](https://user-images.githubusercontent.com/9277616/168450962-32277609-d1d8-4baf-9f31-96f221374163.png)
![P3](https://user-images.githubusercontent.com/9277616/168450963-90b0a462-e698-480a-b0bb-eb153c3b6fc6.png)

**Comment by Xeio** [comment]
Had to reboot the server due to some docker issues and my web server docker didn't come up because the certificate files got deleted. No idea how that happens at the moment... but I restored everything from a backup and it seems to be working now and all the clients are reconnecting.

**Comment by ytzyt** [comment]
No errors right now. Everything seems fine. Thanks!
